csharp_dotnetcore/8.Suggested-Actions/Suggested_Actions/SuggestedActionsBot.cs

```diff
@@ -0,0 +1,121 @@
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Schema;
+
+namespace Suggested_Actions
+{
+    /// <summary>
+    /// This bot will respond to the user's input suggested actions.
+    /// </summary>
+    public class SuggestedActionsBot : IBot
+    {
+        /// <summary>
+        /// This controls what happens when an activity gets sent to the bot.
+        /// </summary>
+        /// <param name="turnContext">Provides the context for the turn of the bot.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A task.</returns>
+        public async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (turnContext == null)
+            {
+                throw new ArgumentNullException(nameof(turnContext));
+            }
+
+            switch (turnContext.Activity.Type)
+            {
+                case ActivityTypes.Message:
+                    var text = turnContext.Activity.Text.ToLowerInvariant();
+                    switch (text)
+                    {
+                        case "red":
+                        {
+                            await turnContext.SendActivityAsync(
+                                "Red is the best color, I agree.",
+                                cancellationToken: cancellationToken);
+                            break;
+                        }
+                        case "yellow":
+                        {
+                            await turnContext.SendActivityAsync(
+                                "Yellow is the best color, I agree.",
+                                cancellationToken: cancellationToken);
+                            break;
+                        }
+
+                        case "blue":
+                        {
+                            await turnContext.SendActivityAsync(
+                                "Blue is the best color, I agree.",
+                                cancellationToken: cancellationToken);
+                            break;
+                        }
+
+                        default:
+                        {
+                            await turnContext.SendActivityAsync(
+                                $"Please select a color from the suggested actions",
+                                cancellationToken: cancellationToken);
+                            break;
+                        }
+                    }
+
+                    await SendSuggestedActionsAsync(turnContext, cancellationToken);
+                    break;
+                case ActivityTypes.ConversationUpdate:
+
+                    // Send a welcome message to the user and tell them what actions they need to perform to use this bot
```

Comment:
Good comment. Needs a "." at the end. 
Also, "need to perform" -> "may perform"